core/src/HIP/Kokkos_HIP_Team.hpp

```diff
@@ -125,22 +125,37 @@ class HIPTeamMember {
 
   KOKKOS_INLINE_FUNCTION int league_rank() const { return m_league_rank; }
   KOKKOS_INLINE_FUNCTION int league_size() const { return m_league_size; }
-  KOKKOS_INLINE_FUNCTION int team_rank() const { return threadIdx.y; }
+  KOKKOS_INLINE_FUNCTION int team_rank() const {
+#ifdef __HIP_DEVICE_COMPILE__
+    return threadIdx.y;
+#else
+    return 0;
```

Comment:
Shouldn't we throw some kind of error if this path is hit at runtime (same below)?

Reply:
This code matches CUDA (see https://github.com/kokkos/kokkos/blob/master/core/src/Cuda/Kokkos_Cuda_Team.hpp#L141) so if an error check is needed it would be needed for both CUDA and HIP.

Reply:
This is a good question but I will merge this PR as is in the hope it can make it into release 3.2